### PR TITLE
Resolves #725, Create Command "/tagclean".

### LINKED ISSUE #725: /tagclean
Would be similar to /nickclean or /nc.

Cleans tags of illegal characters.

### PR DESCRIPTION
**Pull request by ghost**
Resolves #725 Created command "/tagclean" for cleaning forbiiden things from tags, same as the nickclean.
- Testing: Done, Worked fine 

DUFC_Liam's PR is closed so i maked a new one.

### CODE REVIEW

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+public class Command_tagclean extends TFM_Command
+{
+    private static final ChatColor[] BLOCKED = new ChatColor[]
+    {
+        ChatColor.MAGIC,
+        ChatColor.STRIKETHROUGH,
+        ChatColor.ITALIC,
+        ChatColor.UNDERLINE,
+        ChatColor.BLACK
+    };
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: This doesnt look right...

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            final TFM_PlayerData playerdata = TFM_PlayerData.getPlayerData(player);
+            final String Tag = playerdata.getTag();
+            if (Tag != null && !Tag.isEmpty() && !Tag.equalsIgnoreCase(playerName))
+            {
+                final Matcher matcher = REGEX.matcher(Tag);
+                if (matcher.find())
+                {
+                    final String newTag = matcher.replaceAll("");
+                    playerMsg(ChatColor.RESET + playerName + ": \"" + Tag + ChatColor.RESET + "\" -> \"" + newTag + ChatColor.RESET + "\".");
+
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: Why the blank line/

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            final String playerName = player.getName();
+            final TFM_PlayerData playerdata = TFM_PlayerData.getPlayerData(player);
+            final String Tag = playerdata.getTag();
+            if (Tag != null && !Tag.isEmpty() && !Tag.equalsIgnoreCase(playerName))
+            {
+                final Matcher matcher = REGEX.matcher(Tag);
+                if (matcher.find())
+                {
+                    final String newTag = matcher.replaceAll("");
+                    playerMsg(ChatColor.RESET + playerName + ": \"" + Tag + ChatColor.RESET + "\" -> \"" + newTag + ChatColor.RESET + "\".");
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: I dont understand why you are messaging the player here... I cannot see this being at all useful...

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            final String playerName = player.getName();
+            final TFM_PlayerData playerdata = TFM_PlayerData.getPlayerData(player);
+            final String Tag = playerdata.getTag();
+            if (Tag != null && !Tag.isEmpty() && !Tag.equalsIgnoreCase(playerName))
+            {
+                final Matcher matcher = REGEX.matcher(Tag);
+                if (matcher.find())
+                {
+                    final String newTag = matcher.replaceAll("");
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: Maybe call this clearTag or something like that? (Or clearedTag)??

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java (hunk 3 above)
Comment (Adamx120) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: Messaging, to know who need to be smited, cause its a rulebreak

Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: But you look to be messaging the player their name, their original tag and their new tag???

Comment (Adamx120) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: Much easier, than find out the players' new tags :)

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java (hunk 4 above)
Comment (Adamx120) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: Changed.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java (hunk 1 above)
Comment (Adamx120) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: What Line do you think?

Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: I am referring to the ; It doesnt look like it should be there, I am 99% sure that couldnt compile.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java (hunk 3 above)
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: I know, what I am getting at is unless I missed something, you are messaging the player, not the online admins, the player.

**Comment by OxLemonxO** [comment]
I think we should be updating TFM 5.0. not TFM 4.3

**Comment by Wild1145** [comment]
@OxLemonxO Nice spot... I think @JeromSar can fix that if he decides to merge anyway.

**Comment by Adaptified** [comment]
@Wild1145 That's not possible, the server version and repo branch aren't even. Can't contribute as it might disrupt other code.

**Comment by Wild1145** [comment]
@Adaptified Actually that is right, I had forgot that we are not pushing the changes but the entire file. My mistake.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java (hunk 1 above)
Comment (DUFCLiam) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: @Wild1145 It does not show an error in NetBeans...

Comment (Adamx120) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java]: For me it compiles :100:%

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            final TFM_PlayerData playerdata = TFM_PlayerData.getPlayerData(player);
+            final String Tag = playerdata.getTag();
+            if (Tag != null && !Tag.isEmpty() && !Tag.equalsIgnoreCase(playerName))
+            {
+                final Matcher matcher = REGEX.matcher(Tag);
+                if (matcher.find())
+                {
+                    final String cleanTag = matcher.replaceAll("");
+                    playerMsg(ChatColor.RESET + playerName + ": \"" + Tag + ChatColor.RESET + "\" -> \"" + cleanTag + ChatColor.RESET + "\".");
+
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java:44]: Random whitespace?

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                if (matcher.find())
+                {
+                    final String cleanTag = matcher.replaceAll("");
+                    playerMsg(ChatColor.RESET + playerName + ": \"" + Tag + ChatColor.RESET + "\" -> \"" + cleanTag + ChatColor.RESET + "\".");
+        
+                    TFM_PlayerData.getPlayerData(player).setTag(cleanTag);
+                }
+            }
+        }
+
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagclean.java:49]: Again, Random white space?

**Review by Wild1145 — CHANGES_REQUESTED** [review]
Two very minor changes need to be made, and then testing.

**Pull request closed without merging**